### ISSUE TITLE
[webUI] Get execution nodes from API

### ISSUE THREAD

**Issue by kvndrsslr** [dice-group/deer#10]
Right now, execution nodes (operators, readers, writers) are hardcoded in their own files, but we need to generate them dynamically from data we request from the server.
- [ ] As a first step, please familiarize yourself with SHACL [1] [2].
- [ ] Then show that we can generate a execution node in the webUI using only a SHACL shape describing its parameters.

[1]: https://www.w3.org/TR/shacl/
[2]: https://youtu.be/_i3zTeMyRzU

**Comment by kvndrsslr** [comment]
Can you give us an update on the state of this issue please?

**Comment by melissadas** [comment]
Yes, I couldn't push the code relating to this issue since it's not completed in some places. I have been looking at the resources that you provided and I might have some doubts and issues. I will sort it down and then talk in a meeting with you to clear my doubts.

**Comment by kvndrsslr** [comment]
FYI, code doesn't need to be complete in order to push it - otherwise nobody would ever push anything ;)

**Comment by kvndrsslr** [comment]
1. Take some shape 
2. Hardcode shape into frontend
3. Modify shapes and use the shacl.js lib to make progress

**Comment by kvndrsslr** [comment]
Please code a factory function to create all nodes dynamically

**Comment by pskLana** [comment]
Now nodes are created dynamically based only on a SHACL shape.
![image](https://user-images.githubusercontent.com/48616148/99881137-19e1d580-2c18-11eb-8e01-ee25af3c05fc.png)

**Comment by kvndrsslr** [comment]
We can close this once everything is dynamic. As far as I understood, the right panel still remains to be done.

**Comment by pskLana** [comment]
The right panel is also generated dynamically. We just need to fill it with proper fields depending on SHACL file. Although, yes, we can keep it until it is done.

**Comment by kvndrsslr** [comment]
Dynamic generation basics are implemented
